fairlearn/post_processing/roc_curve_based_post_processing.py

```diff
@@ -360,42 +380,89 @@ def _roc_curve_based_post_processing_equalized_odds(attributes, labels, scores,
 
 def _vectorized_prediction(function_dict, A, scores):
     """ Make predictions for all samples with all provided functions,
-    but use only the results from the function that corresponds to the protected
+    but use only the results from the function that corresponds to the sensitive
     attribute value of the sample.
 
-    :param function_dict: the functions that apply to various protected attribute values
+    :param function_dict: the functions that apply to various sensitive attribute values
     :type function_dict: dictionary of functions
-    :param A: protected attributes for each sample
+    :param A: sensitive attributes for each sample
     :type A: vector as Series or ndarray
     :param scores: vector of predicted values
     :type scores: vector as Series or ndarray
     """
-    A_vector = A
-    scores_vector = scores
-
-    if type(A_vector) == list:
-        A_vector = np.array(A)
-
-    if type(scores_vector) == list:
-        scores_vector = np.array(scores)
+    # handle type conversion to ndarray for other types
+    A_vector = _convert_to_ndarray(A, MULTIPLE_AUX_DATA_COLUMNS_ERROR_MESSAGE)
+    scores_vector = _convert_to_ndarray(scores, SCORES_DATA_TOO_MANY_COLUMNS_ERROR_MESSAGE)
 
     if len(A_vector) != len(scores_vector):
-        raise ValueError("The protected attribute vector needs to be of the same length as the "
-                         "scores vector.")
+        raise ValueError(DIFFERENT_INPUT_LENGTH_ERROR_MESSAGE.format("A and scores"))
 
     return sum([(A_vector == a) * function_dict[a].predict(scores_vector) for a in function_dict])
 
 
-def _sanity_check_and_group_data(attributes, labels, scores):
+def _convert_to_ndarray(data, dataframe_multiple_columns_error_message):
+    if type(data) == list:
+        data = np.array(data)
+    elif type(data) == pd.DataFrame:
+        if len(data.columns) > 1:
+            # TODO: extend to multiple columns for additional data
+            raise ValueError(dataframe_multiple_columns_error_message)
+        data = data[data.columns[0]].values
```

Comment:
Why not just grab `values` generally from the DataFrame?

Reply:
I started out thinking about this for the future when we'll have multiple attributes, at which point I want to extract them separately (because there could be some in X, some in auxiliary data) so that'll be the way to go. Making it fully compatible turned out to be too much work for right now, but probably sometime in November or so I'll get to it.

Reply:
Oh there was one more consideration. If I directly get the values it's of shape (n,1), whereas if I get it this way it's shape (3,)